### ISSUE TITLE
Code does not update after deploy 

### ISSUE THREAD

**Issue by truonglocbinh** [aws-amplify/amplify-hosting#3504]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting questions](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

dq4gdul9tuikh

### AWS Region

us-east-1

### Amplify Hosting feature

Frontend builds

### Question

I have a Vue3 project  use default amplify to build. I saw that everytime I update my code and deploy it, but my site does not update right away, I need clear browser cache manually to get new version. It is hard for my customer because I can not ask them clear browser cache everytime I update site. When I use vps to host I don't see it although the file name is change with new hash . Is there any way to make it clear cache after deploy ?

**Comment by nadetastic** [comment]
Hi @truonglocbinh thank you for opening this issue - to clarify is it the data (from an api for example) that is not updating right away or the site itself ex the html?

**Comment by truonglocbinh** [comment]
@nadetastic The site itself, example, when I add a button but after deploy , I dont see new button I added, but after I try clear browser cache I can see it

**Comment by nadetastic** [comment]
@truonglocbinh thanks for confirming. A few more questions, since this seems to be resolved by refreshing the browser cache, are you able reproduce the problem with a different browser or different device/computer?

Additionally, i see that this app has a custom domain, do you notice the issue on both the custom domain as well as the default amplify generated domain?

**Comment by truonglocbinh** [comment]
@nadetastic Yes it raise with other browser, and need to be clear cache , the problem is when i deploy I can not ask my customer clear cache everytime. I saw both default and custom domain same issue

**Comment by truonglocbinh** [comment]
Any update for that ?, Maybe service worker cache that ?

**Comment by truonglocbinh** [comment]
I try to create example and has same problem 
https://github.com/truonglocbinh/vite
https://main.d31vqfozggoynx.amplifyapp.com/

**Comment by truonglocbinh** [comment]
I deploy on 06/06/2023 but it return a version from 30/05/2023 
![Screen Shot 2023-06-06 at 15 07 51](https://github.com/aws-amplify/amplify-hosting/assets/8655190/4287a73a-1edc-4a12-a3b9-dd1e33da69b8)
om 30/05/2023

**Comment by multivoltage** [comment]
Same problem

**Comment by nadetastic** [comment]
Hi @multivoltage 

Could you share your app id/url so i can test the same?

**Comment by multivoltage** [comment]
@nadetastic I wrote here all info https://github.com/aws-amplify/amplify-hosting/issues/3516
in my case Date and Last modified are different.
I also noticed that many people has this bug

**Comment by truonglocbinh** [comment]
@nadetastic I deployed new version , could you check it ?

**Comment by espenbye** [comment]
Same issue with Next.js application. Have deployed several times today, but is still being servered a cache version.

![image](https://github.com/aws-amplify/amplify-hosting/assets/106824/ff31ad10-8b2d-4c6c-8478-9f249533c670)

**Comment by multivoltage** [comment]
For all, please check you env variables. 
besides this problem, I noticed that Amplify load wrong variables env (in my case production take staging envs). I noticed only in production :), evenf if in "env variables view" was all good

**Comment by truonglocbinh** [comment]
@multivoltage Which is env  and how you know it is wrong while env variables view" was all good ?

**Comment by multivoltage** [comment]
I know because I have intercom widget with some id in production and intercom widget with differnt id in staging. 
Some clients send email with chat support to test account :). 

Initially I thuth that a bad configuration was createt inside amplify, but looking at "Environment Variables" it was all ok

**Comment by truonglocbinh** [comment]
@multivoltage it hard to check with env while you get a lucky chance from clients

**Comment by nadetastic** [comment]
@truonglocbinh from my end it appears I see the latest deployment

![Screenshot 2023-06-08 at 2 47 52 PM](https://github.com/aws-amplify/amplify-hosting/assets/87148442/aeb96236-1434-4285-ac0a-b76001b4fcf5)

**Comment by truonglocbinh** [comment]
I think it will expire in a moment time, my purpose is how to clear it after deploy

**Comment by multivoltage** [comment]
Have you got custom redirect to form cloudfront?

**Comment by truonglocbinh** [comment]
All as default I think

**Comment by truonglocbinh** [comment]
I found a different when I open new chrome Profile, I see it does not get `https://main.d31vqfozggoynx.amplifyapp.com/manifest.webmanifest` but when I get my site on old chrome Profile, it will fetch `https://main.d31vqfozggoynx.amplifyapp.com/manifest.webmanifest` and I can not see new code deployed @hloriii @nadetastic

**Comment by hloriii** [comment]
Hi @truonglocbinh 👋🏽 thanks for your patience here. Try adding `must-revalidate` to your `Cache-Control` header and see if that resolves the issue for you. Ex: `Cache-Control: max-age=0, must-revalidate`.

**Comment by hloriii** [comment]
@KevenSoares22 if you're still running into this issue, please share your app ID. Also, what frontend framework are you using for your application?
